src/pages/EarthEngineImport/components/BandCocMappingTable.js

```diff
@@ -14,61 +14,36 @@ import {
 import React, { useState, useEffect } from 'react'
 import { FieldArray } from 'react-final-form-arrays'
 import { useCachedDataQuery } from '../util/CachedQueryProvider.js'
-import {
-    getEarthEngineBands,
-    POPULATION_AGE_GROUPS_DATASET_ID,
-} from '../util/earthEngines.js'
+import { getEarthEngineBands } from '../util/earthEngines.js'
 import { BAND_COCS } from '../util/formFieldConstants.js'
 
 const { Field, useFormState, useForm } = ReactFinalForm
 
-const MappingTable = () => {
+const BandCocMappingTable = () => {
     const { values } = useFormState()
-    const form = useForm()
+    const { change } = useForm()
     const { dataElements } = useCachedDataQuery()
     const [cocs, setCocs] = useState([])
     const { earthEngineId, dataElementId, bandCocs } = values
 
-    const { push, update, pop } = form.mutators
-
     useEffect(() => {
-        getEarthEngineBands(POPULATION_AGE_GROUPS_DATASET_ID).forEach((band) =>
-            push(BAND_COCS, {
-                bandId: band.id,
-                bandName: band.name,
-            })
-        )
-
+        //clear form bandCocs when this component is removed
         return function cleanup() {
-            const numBandCocs = getEarthEngineBands(
-                POPULATION_AGE_GROUPS_DATASET_ID
-            ).length
-
-            for (let i = 0; i < numBandCocs; ++i) {
-                pop(BAND_COCS)
-            }
+            change(BAND_COCS, [])
         }
-    }, []) // eslint-disable-line react-hooks/exhaustive-deps
+    }, [change])
 
     useEffect(() => {
-        bandCocs &&
-            bandCocs.forEach((bc, index) => {
-                update(BAND_COCS, index, {
-                    bandId: bc.bandId,
-                    bandName: bc.bandName,
-                })
-            })
-        if (dataElementId) {
-            const newCocs = dataElements.find(({ id }) => id === dataElementId)
-                .categoryCombo.categoryOptionCombos
+        const dataSetBands = getEarthEngineBands(earthEngineId)
+        change(BAND_COCS, dataSetBands)
+    }, [earthEngineId, dataElementId, change])
```

Comment:
That's a good point. It is needed because if it changes, then the cocs on the bands are no longer valid. BAND_COCS is a map of the earth engine band (e.g Female 5 - 9 yr) to a dhis2 coc connected to the active Data Element.  Not sure what convention says about dependencies that aren't actually used inside the effect.

Reply:
I'll add a comment to the code about this.